src/main/java/software/amazon/event/kafkaconnector/logging/GitCommitAwareLoggerFactory.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.event.kafkaconnector.logging;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.Properties;
+import org.apache.kafka.common.utils.LogContext;
+import org.slf4j.Logger;
+
+public class GitCommitAwareLoggerFactory {
+
+  private static final LogContext context;
+
+  static {
+    var properties = new Properties();
+    try {
+      properties.load(
+          Objects.requireNonNull(GitCommitAwareLoggerFactory.class.getResource("/git.properties"))
```

Comment:
question: `getResource` uses `/` root path syntax for class resource files? Or would it work also without `/` or even `./...`? Just curious.

Reply:
Tried w/o `/` and it does not work. I guess it's the root of the jar file.

Reply:
We can leverage try with resources here to automatically close the stream:

``` java
try (var something = Something.class.getClassLoader().getResourceAsStream("properties")) {
    var properties = new Properties();

    properties.load(something);

    //etc.
} catch (IOException ex) {
    System.out.println("Something");
}
```

Reply:
Good catch!